### ISSUE TITLE
KeyError: 'stats' in player.py

### ISSUE THREAD

**Issue by MihaMuskinja** [cwendt94/espn-api#603]
### Sport

Basketball

### Summary

I get a KeyError: 'stats' during the api execution. Has something changed in the api?

### Logs/Data

```
  File "/Users/mihamuskinja/git/fantasy/api.py", line 56, in get_league
    league = League(league_id=109755, year=2025, swid=swid, espn_s2=espn_s2, debug=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/basketball/league.py", line 19, in __init__
    self.fetch_league()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/basketball/league.py", line 23, in fetch_league
    self._fetch_teams(data)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/basketball/league.py", line 50, in _fetch_teams
    super()._fetch_teams(data, TeamClass=Team, pro_schedule=pro_schedule)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/base_league.py", line 86, in _fetch_teams
    self.teams.append(TeamClass(team, roster=roster, schedule=schedule, year=seasonId, owners=owners, pro_schedule=pro_schedule))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/basketball/team.py", line 30, in __init__
    self._fetch_roster(roster, year, kwargs.get('pro_schedule'))
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/basketball/team.py", line 44, in _fetch_roster
    self.roster.append(Player(player, year, pro_schedule))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniconda/base/envs/fantasy/lib/python3.11/site-packages/espn_api/basketball/player.py", line 45, in __init__
    if split['stats']:
       ~~~~~^^^^^^^^^
KeyError: 'stats'
```

**Comment by monsdar** [comment]
Seems to be similar to this issue: https://github.com/cwendt94/espn-api/pull/591/files

So changing `split['stats']` to `split.get('stats')` should be good enough.

**Comment by monsdar** [comment]
Yeah, that works fine for me locally, PR incoming

**Comment by martincham** [comment]
Was curious about why this is happening. Heres some splits for Bogdan Bogdanovich. Some werid stuff going on with the 'externalId' as well.
<img width="310" alt="Screenshot 2024-10-31 at 10 13 14 AM" src="https://github.com/user-attachments/assets/83a4f153-ad0c-4a97-ad00-a5c55c01ac60">
<img width="262" alt="Screenshot 2024-10-31 at 10 13 21 AM" src="https://github.com/user-attachments/assets/536d32a7-99d9-447b-af5b-22b05a6190f6">
<img width="264" alt="Screenshot 2024-10-31 at 10 13 28 AM" src="https://github.com/user-attachments/assets/60e315bf-593c-449e-85d8-5f81f0f88bc5">

**Comment by atagunduzalp** [comment]
hi guys, when is this commit merged and usable? thanks!

**Comment by shaggycamel** [comment]
I'm experiencing this issue too

**Comment by nathanlochunho** [comment]
Same, hope this issue will be fixed soon!

**Comment by monsdar** [comment]
Even when my PR fixes the error it's still unclear what happens datawise. It's not really a fix when there's no exception anymore but half of the data is missing.

**Comment by martincham** [comment]
>  It's not really a fix when half of the data is missing.

There doesn't seem to be data missing. This case is only happening for Bogdan Bogdaovich, Dejounte Murray, and Immanuel Quickly. They all played the 1st game of the season and have missed since, so their "Last 7" stats are empty. For some reason ESPN is populating these players differently, but there's no data missing. Perhaps some edge-case error they introduced in their API?

**Comment by martincham** [comment]
> Same, hope this issue will be fixed soon!





> hi guys, when is this commit merged and usable? thanks!


You can also fix it locally. Go to player.py, or debug and trace it to where your player.py is located, and change line 45 to :

`if split.get('stats')`

**Comment by monsdar** [comment]
> There doesn't seem to be data missing.

Awesome, thanks for checking that!

> You can also fix it locally.

I just ported our Telegram bot to run as an Azure Function. There's no easy way to fix it within the function, so I guess I'll need to wait for the merge. Hooray, Cloud!

**Comment by cwendt94** [comment]
Thanks for all the investigation! The fix has been added in v0.41.0!
